optuna/pruners/threshold.py

```diff
@@ -71,12 +83,12 @@ def __init__(
         interval_steps: int = 1,
     ) -> None:
 
-        if lower is not None and not isinstance(lower, float):
-            raise TypeError("lower should be either None or a floating point.")
-        if upper is not None and not isinstance(upper, float):
-            raise TypeError("upper should be either None or a floating point.")
         if lower is None and upper is None:
-            raise ValueError("Either lower or upper must be specified.")
+            raise TypeError("Either lower or upper must be specified.")
+        if lower is not None:
+            _check_value(lower)
+        if upper is not None:
+            _check_value(upper)
```

Comment:
How about returning and reassigning the casted value?

Reply:
713eda8 :+1: